app/fee/fee_util.go

```diff
@@ -0,0 +1,168 @@
+package fee
+
+import (
+	"math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+
+	oraclekeeper "github.com/bandprotocol/chain/v2/x/oracle/keeper"
+	"github.com/bandprotocol/chain/v2/x/oracle/types"
+)
+
+// getTxPriority returns a naive tx priority based on the amount of the smallest denomination of the gas price
+// provided in a transaction.
+// NOTE: This implementation should be used with a great consideration as it opens potential attack vectors
+// where txs with multiple coins could not be prioritize as expected.
+func getTxPriority(fee sdk.Coins, gas int64) int64 {
+	var priority int64
+	for _, c := range fee {
+		p := int64(math.MaxInt64)
+		// multiplied by 1000 first because priority is int64.
+		// otherwise, if gas_price < 1, the priority will be 0.
+		gasPrice := c.Amount.MulRaw(1000).QuoRaw(gas)
+		if gasPrice.IsInt64() {
+			p = gasPrice.Int64()
+		}
+		if priority == 0 || p < priority {
```

Comment:
if there 0 coin or <0.001 coin the priority will be zero which will be set again

Reply:
True, already updated the logic to use priority based on gas price of uband unit.